Form/Builder.php

```diff
@@ -0,0 +1,319 @@
+     * @param string $defaultStructureView
+     */
+    public function __construct(
+        RequestStack $requestStack,
+        FormRepository $formRepository,
+        CollectionStrategyInterface $collectionStrategy,
+        FormFactory $formFactory,
+        $defaultStructureView
+    ) {
+        $this->requestStack = $requestStack;
+        $this->formRepository = $formRepository;
+        $this->collectionStrategy = $collectionStrategy;
+        $this->formFactory = $formFactory;
+        $this->defaultStructureView = $defaultStructureView;
+    }
+
+    public function buildByRequest(Request $request)
+    {
+        foreach ($request->request->all() as $key => $parameters) {
+            if (strpos($key, 'dynamic_') === 0) {
+                $formNameParts = explode('dynamic_', $key, 2);
+
+                if (!isset($formNameParts[1])) {
+                    continue;
+                }
+
+                $name = $formNameParts[1];
+                $locale = $request->getLocale();
+
+                $structure = $request->attributes->get('structure');
+
+                if (
+                    !$structure instanceof StructureInterface
+                    || !$structure->hasProperty('title')
+                    || !$structure->hasProperty($name)
+                ) {
+                    continue;
+                }
+
+                $typeId = $structure->getUuid();
+                $typeName = $structure->getProperty('title')->getValue();
+                $id = (int) $structure->getProperty($name)->getValue();
+
+                if (!$typeId || !$id || !$typeName) {
+                    continue;
+                }
+
+                return $this->build($id, 'page', $typeId, $typeName, $locale, $name);
+            }
+        }
+
+        return [null, null];
+    }
+
+    public function build($id, $type, $typeId, $typeName, $locale = null, $name = 'form')
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (!$locale) {
+            $locale = $request->getLocale();
+        }
+
+        $key = $this->getKey($id, $type, $typeId, $typeName, $locale, $name);
+
+        // Load Form entity
+        $formEntity = $this->loadFormEntity($id, $locale);
+
+        if (!$formEntity) {
+            return [null, null];
+        }
+
+        $webspaceKey = $this->getWebspaceKey();
+        $defaults = $this->getDefaults($formEntity, $locale);
+
+        // Create Form Type
+        $formType = $this->createFormType(
+            $formEntity,
+            $locale,
+            $name,
+            $type,
+            $typeId,
+            $typeName
+        );
+
+        // Create Form
+        $form = $this->createForm(
+            $formType,
+            $typeId,
+            $locale,
+            $formEntity,
+            $webspaceKey,
+            $defaults
+        );
+
+        // Handle request
+        $form->handleRequest($request);
+
+        return [$formType, $form];
```

Comment:
type currently stores the data for email which currently is needed before the form is submitted.
In the refractoring for sulu form bundle the `AbstractType` need to be removed (https://github.com/sulu/SuluFormBundle/issues/8). It can get the needed information directly from `$dynamic->getForm()->...` or from a configuration.